tmail-backend/tmail-third-party/openpaas/src/test/java/com/linagora/tmail/contact/OpenPaasContactsConsumerTest.java

```diff
@@ -0,0 +1,378 @@
+        indexJhonDoe(OpenPaasServerExtension.ALICE_EMAIL());
+
+        sendMessage("""
+            {
+                "bookId": "ALICE_USER_ID",
+                "bookName": "contacts",
+                "contactId": "fd9b3c98-fc77-4187-92ac-d9f58d400968",
+                "userId": "ALICE_USER_ID",
+                "vcard": [
+                "vcard",
+                  [
+                     [ "version", {}, "text", "4.0" ],
+                     [ "kind",    {}, "text", "individual" ],
+                     [ "fn",      {}, "text", "Jhon Dont" ],
+                     [ "email",   {}, "text", "jhon@doe.com" ],
+                     [ "org",     {}, "text", [ "ABC, Inc.", "North American Division", "Marketing" ] ]
+                  ]
+               ]
+            }
+            """);
+
+        ContactFields expectedContact =
+            new ContactFields(new MailAddress("jhon@doe.com"), "Jhon Dont", "");
+
+        await().timeout(TEN_SECONDS).untilAsserted(() ->
+            assertThat(
+                Flux.from(searchEngine.autoComplete(AccountId.fromString(OpenPaasServerExtension.ALICE_EMAIL()), "jhon", 10))
+                    .collectList().block())
+                .hasSize(1)
+                .map(EmailAddressContact::fields)
+                .first()
+                .isEqualTo(expectedContact));
+    }
+
+    @Test
+    void givenDisplayNameFromOpenPaasIsEmptyThenStoredDisplayNameShouldPersist() {
+        ContactFields indexedContact = indexJhonDoe(OpenPaasServerExtension.ALICE_EMAIL());
+
+        sendMessage("""
+            {
+                "bookId": "ALICE_USER_ID",
+                "bookName": "contacts",
+                "contactId": "fd9b3c98-fc77-4187-92ac-d9f58d400968",
+                "userId": "ALICE_USER_ID",
+                "vcard": [
+                "vcard",
+                  [
+                     [ "version", {}, "text", "4.0" ],
+                     [ "kind",    {}, "text", "individual" ],
+                     [ "fn",      {}, "text", "  " ],
+                     [ "email",   {}, "text", "jhon@doe.com" ],
+                     [ "org",     {}, "text", [ "ABC, Inc.", "North American Division", "Marketing" ] ]
+                  ]
+               ]
+            }
+            """);
+
+        await().timeout(TEN_SECONDS).untilAsserted(() ->
+            assertThat(
+                Flux.from(searchEngine.autoComplete(AccountId.fromString(OpenPaasServerExtension.ALICE_EMAIL()), "jhon", 10))
+                    .collectList().block())
+                .hasSize(1)
+                .map(EmailAddressContact::fields)
+                .first()
+                .isEqualTo(indexedContact));
+    }
+
+    /*
+    * The automatic contact indexing is triggered when you send or receive a message
+    * from someone, then their contact info be automatically indexed in the contacts' search engine.
+    * */
+    @Test
+    void automaticContactIndexingShouldNotOverrideContactInfoFromOpenPaas() {
+        //
+    }
```

Comment:
?

Reply:
Now that we have defined the expected behavior. i.e. OpenPaas is the single source of truth for contacts, this test case become irrelevant.